[BugFix] Load Problem for generated column (#25828)

srlch opened the pull request:
Problem:
1. Streamload and Broker load will failed if generated column
ref some mapping column
2. spark load fail

Solution:
1. remove the restriction and allow the generated column ref the 
mapping column
2. add the generated column column desc when planning spark load job

Fixes #25828

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 15 (26.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkLoadPendingTask.java|1|4|25.00%|[536, 537, 538]|
|:large_blue_circle:|com/starrocks/load/Load.java|3|11|27.27%|[343, 344, 345, 346, 347, 348, 349, 351]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 22 (22.73%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/Load.java|4|18|22.22%|[243, 244, 245, 246, 247, 248, 345, 346, 347, 348, 349, 350, 351, 353]|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkLoadPendingTask.java|1|4|25.00%|[536, 537, 538]|

srlch committed:
[BugFix] Load Problem for generated column (#25828)

Problem:
1. Streamload and Broker load will failed if generated column
ref some mapping column
2. spark load fail

Solution:
1. remove the restriction and allow the generated column ref the
mapping column
2. add the generated column column desc when planning spark load job

Signed-off-by: srlch <linzichao@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 21 / 56 (37.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/SparkLoadPendingTask.java|1|4|25.00%|[536, 537, 538]|
|:large_blue_circle:|com/starrocks/load/Load.java|20|52|38.46%|[244, 245, 246, 247, 248, 249, 359, 516, 763, 804, 805, 806, 807, 808, 811, 812, 813, 815, 816, 817, 818, 820, 821, 823, 826, 827, 828, 829, 830, 832, 833, 834]|

wanpengfei-git commented:
@Mergifyio backport branch-3.1
